### ISSUE TITLE
Bad performance when making many queries without waiting for response right after Prisma client is created

### ISSUE THREAD

**Issue by rpominov** [prisma/prisma#15400]
### Bug description

The following code takes ~6000ms to run.

```js
import { PrismaClient } from "@prisma/client";

const main = async () => {
  const prisma = new PrismaClient();

  // await prisma.$queryRawUnsafe("SELECT 1");

  const start = new Date().getTime();

  await Promise.all(
    Array.from({ length: 1000 }).map(() => prisma.$executeRawUnsafe("SELECT 1"))
  );

  console.log(new Date().getTime() - start);
};

main();
```

However, when I uncomment the commented out line, it takes just ~200ms. 

### Expected behavior

Adding `await prisma.$queryRawUnsafe("SELECT 1");` should not affect the performance of the code that goes after.

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["interactiveTransactions"]
}

datasource db {
  provider = "postgres"
  url      = env("DATABASE_URL")
}
```

### Environment & setup

Hardware: MacBook Pro (13-inch, M1, 2020)
OS: MacOS 12.6
Node: v16.16.0
Prisma: 4.1.0
PostgreSQL: 14.1 installed via https://www.postgresql.org/download/macosx/

NODE_ENV=production

### Prisma Version

```
prisma                  : 4.1.0
@prisma/client          : 4.1.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at ../../node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at ../../node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at ../../node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at ../../node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : 8d8414deb360336e4698a65aa45a1fbaf1ce13d8
Studio                  : 0.467.0
```

**Comment by janpio** [comment]
Interesting. Guess: This is caused by all these Promises/queries trying to create a connection at the same time. With the added initial query that seems to be handled much better. Internal note: The tracing results for this will probably be interesting.

**Comment by SevInf** [comment]
I think it might be [the issue](https://github.com/prisma/prisma/issues/14695) we fixed in 4.2.1. I see you are on the older version. Could you update to the latest one and report if you are still experiencing the issue?

**Comment by rpominov** [comment]
Yes. Appears to be fixed in 4.3.1. Thank you!
